distributed/protocol/serialize.py

```diff
@@ -193,9 +197,54 @@ def register_serialization_family(name, dumps, loads):
 register_serialization_family("error", None, serialization_error_loads)
 
 
+def infer_if_recurse_to_serialize_list(x: list) -> bool:
+    """Determine how to serialize lists
+
+    Function sets iterate_collection variable in serialize for
+    lists.  If it returns False, the list will be serialized
+    using Dask Dispatch.  It it returns True, the list will
+    be serialized recursively with pickle
+    """
+    try:
+        iseq = iter(x)
+        first_val = next(iseq)
+        if is_dask_collection(first_val) or typename(type(first_val)) not in [
+            "str",
+            "int",
+            "float",
+        ]:
```

Comment:
Trying to use `isinstance(first_val, (str, int, float))` returns test failures in `shuffle`.

Additionally, this seems more consistent with the convention of using `type(x)`, which is the preferred approach in `serialize.py`.  I could see replacing with `type(x) in [int, float]` if that makes more sense...

Thoughts?

Reply:
This makes me a bit nervous. 
AFAIK, the use of `type(x)` is only here to avoid sub-classes of `list, set, tuple, dict` to be converted to their base-class by msgpack. By using Dispatch on the collection, we make sure to preserve the sub-classes. 
Do you have a reproducer for the shuffle failing? 

I think this is another good reason why we need to re-design and simplify the serialization in Dask.